### ISSUE TITLE
Cannot refresh

### ISSUE THREAD

**Issue by golden-peaches** [XilinJia/Podcini#4]
### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/XilinJia/PodVinci/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/XilinJia/PodVinci/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

4.0.1

### Where did you get the app from

F-Droid

### Android version

Android 14

### Device model

Galaxy S22

### First occurred

1 day ago

### Steps to reproduce

1. I had added a few podcasts via the search function, they added and loaded episodes of the podcasts as it should.
2. I was able to play episodes.
3. I decided to try to export my antennapod database and import into podcini. No error messages occurred, and all my antennapod subscriptions and history were added to podcini.
4. Refresh subscriptions.

### Expected behaviour

I expected that I could refresh and see new episodes.

### Current behaviour

No new episodes load. The refresh loading symbol quickly disappears, quicker than a proper refresh would take to complete. Same behavior on any screen (episodes, subscriptions, or individual podcast page), either pull to refresh or pressing refresh in the dropdown menu.

**Comment by XilinJia** [comment]
Interesting.  I use it on my S21 Android 14, and imported AntennaPod DB.  It does refresh.  Note, on the Subscription view, after the refresh, the list may or may not change depending on your sort setting.  But if you open a podcast and should see the episodes updated.  You don't see that happening?

If you are on mobile network, be sure to set permission in Setting -> Mobile updates.

There is a newer version 4.0.1-1 for title display bug fix, not sure if that would make any difference for you on refresh.

**Comment by golden-peaches** [comment]
I check my sort settings, and everything is set to sort by newest (except my subscriptions listing page). So when double checking if any new episodes were found, there were no new episodes found.

I toggled the Mobile updates setting (even tough I've been on wifi), and no change.

I downloaded the 4.01-1 apk from github and installed. No change in behavior.

I have added two new podcasts via the search function. They do load all episodes for the newly added subscriptions. In particular even episodes after the date I did the database import.

**Comment by XilinJia** [comment]
With the not-refreshed podcasts, are there errors?  On the subscriptions view there should a red dot, or in the podcast view, there should be a red bar below the image.  tap on the red bar, there should be some info related to the error.

**Comment by golden-peaches** [comment]
To answer your question: no errors appear, nor any change in the listings.

I looked at my settings in the app, and I found one partial solution.

I had not allowed notifications in my system settings for podcini. Allowing notifications, refresh works as normal and seems to find all episodes, except for 3 of my subscriptions. However, unlike before there is now an error message "Last refresh failed. Tap to view details". The details read "IO error" for all three subscriptions (Frau Feller & Frau Jahnke, This American Life, and Hoss  & Hopf). Only for the subscription "This American Life", in the error details the technical reason is "getValue(...) must not be null".

**Comment by XilinJia** [comment]
can you get me some more detail on "getValue(...) must not be null",  or perhaps a screenshot of the popup details?

**Comment by golden-peaches** [comment]
![Screenshot_20240227_223621_Podcini](https://github.com/XilinJia/Podcini/assets/120830230/249ecdc2-37c8-42b2-ab86-008b97284177)

**Comment by golden-peaches** [comment]
I guess from the time I installed Podcini, I had notifications for Podcini off. (Notifications for my whole phone are on.) It does look like toggling the notifications permission for Podcini is the main issue causing the refresh not to work.

**Comment by XilinJia** [comment]
I checked about this, but wan't able to reproduce.
@ValiumBear: which Podcini version do you use? you also use it on S22 or Android 14?  Did the issue also occur when you set notification off?

**Comment by XilinJia** [comment]
It appears having to do with the top level notification permission in the setting of Podcini.  Once the top level is enabled, even though all other sub-categories are not, refresh works.  If the top level is disabled, then refresh doesn't happen.
Will check further.

**Comment by XilinJia** [comment]
Since Android 13 (or SDK 33) permission POST_NOTIFICATIONS is required for (among other thing) running foreground service like an AV player.   After I removed Home fragment, the permission is not requested to the user, and in the download worker if the permission is not granted, regardless of Android version, it breaks out (this part of code is added after Antenna 3.2.0 release, and I forked an unstable code base).

Now a permission request is added in the main activity for it.
